### ISSUE TITLE
asm_thumb bug: wrong restriction of integer range in data(4, ...) statement

### ISSUE THREAD

**Issue by rkompass** [micropython/micropython#8956]
The following code illustrates the issue:

```
@micropython.asm_thumb
def ret_num():
        nop()                  # for alignment
        data(2, 0x4800)        # ldr  r0, [pc, #0]  load r0 with data following the branch
        b(HERE)
        data(4, 0x40000000)    # r0 = 0x40000000  -> SyntaxError: 'data' requires integer arguments; r0 = 0x3fffffff still works
        label(HERE)

print(hex(ret_num()))
```
should return 1073741824 which is 0x40000000 but instead throws the exception
  SyntaxError: 'data' requires integer arguments.
With 0x3fffffff and smaller values in the data(4, ...) statement there is no exception.
I assume that here a wrong test is performed which tests against the limit of internal representation which is 2^30 instead of
the 32 bit limit which is 2^32.

**Comment by dpgeorge** [comment]
Thanks for the report.  Should be fixed by #8959.

**Comment by dpgeorge** [comment]
Fixed by c0fa903d6b2ed5131ae60f8faff2c6ad5276b3a2
